broker/src/main/java/io/camunda/zeebe/broker/transport/backpressure/StabilizingAIMDLimit.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.broker.transport.backpressure;
+
+import com.google.common.base.Preconditions;
+import com.netflix.concurrency.limits.limit.AbstractLimit;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * The limit is calculated purely based on the configured expectedRTT and observed rtt (round trip
+ * time). The algorithm tries to keep the rtts around expectedRTT. It is not guaranteed to be always
+ * less than that. But on an average, rtts will be less than the configured expectedRTT.
+ *
+ * <p>The implementation is based on {@link com.netflix.concurrency.limits.limit.AIMDLimit}.
+ * AIMDLimit has a limitation that the limit fluctuates between 1 and 2*X, where X is the optimal
+ * limit for the system. It rarely stabilizes at X. This results in always fluctuating throughput
+ * and latency. {@link StabilizingAIMDLimit} fixes this issue, by not reducing the limit if the
+ * inflight is greater than the current limit. As a result it attempts to keep the limit around X.
```

Comment:
:question: was that something we came up ourselves or is this based on something we have read  or reused ? :thinking:

Reply:
It was my observation from my evaluation when we first introduced the backpressure. [Here](https://docs.google.com/presentation/d/1vCWDqssu5lfq81gGe3F--RpCpVsqwB2LXldBQ0YV6Ag/edit#slide=id.g84efc3e11a_0_67) is some results shared in one tech talk.